artiq/test/test_scheduler.py

```diff
@@ -90,7 +90,7 @@ def setUp(self):
 
     def test_steps(self):
         loop = self.loop
-        scheduler = Scheduler(_RIDCounter(0), dict(), None)
+        scheduler = Scheduler(_RIDCounter(0), dict(), None, "experiment_submissions.csv")
```

Comment:
If you're not reading the file in the test, then don't write it and set it to ``None``. You could just use ``None`` as default value in the ``Scheduler``'s parameter, then it's backward-compatible and you don't need to edit the test and potentially other code.